hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java

```diff
@@ -325,6 +325,32 @@ private void buildApiNodeEdges(StructuredTraceGraph graph) {
           }
         }
       }
+      // Sometimes an exit span might be missing for services like Istio, Kong.
+      // Only Entry spans will be populated for these services,
+      // an edge must be created between these services as well.
+      Optional<Event> entryBoundaryEvent = apiNode.getEntryApiBoundaryEvent();
+      if (entryBoundaryEvent.isPresent()) {
+        List<Event> children = graph.getChildrenEvents(entryBoundaryEvent.get());
+        if (children != null) {
+          for (Event child : children) {
+            // if the child of an entry boundary event is an entry api boundary type,
+            // which can happen if exit span missing and both belongs to different services.
+            if (EnrichedSpanUtils.isEntryApiBoundary(child)
+                && GraphBuilderUtil.areBothSpansFromDifferentService(
+                    child, entryBoundaryEvent.get())) {
+              ApiNode<Event> destinationApiNode = entryApiBoundaryEventIdToApiNode.get(child);
+              LOGGER.debug(
+                  "Edge between entry boundaries {} to {} ",
```

Comment:
Add some more info in the debug logs - span ids, trace id

Reply:
done